aws/resource_aws_ec2_tag_gen_test.go

```diff
@@ -0,0 +1,73 @@
+// Code generated by internal/tagresource/generator/main.go; DO NOT EDIT.
+
+package aws
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/tagresource"
+)
+
+func testAccCheckEc2TagDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).ec2conn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_ec2_tag" {
+			continue
+		}
+
+		identifier, key, err := tagresource.GetResourceId(rs.Primary.ID)
+
+		if err != nil {
+			return err
+		}
+
+		exists, _, err := keyvaluetags.Ec2GetTag(conn, identifier, key)
+
+		if err != nil {
+			return err
+		}
+
+		if exists {
+			return fmt.Errorf("resource (%s) tag (%s) still exists", identifier, key)
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckEc2TagExists(resourceName string) resource.TestCheckFunc {
```

Comment:
Is there a reason this isn't using the pattern of populating the a resource passed by reference? I know it's unused in these particular tests

Reply:
These `Check...Exists` functions should likely go away altogether across the codebase and the attribute checking from the returned API object is rarely needed, given the SDK state value checking functions are robust enough nowadays. This testing is covered by the automatic refresh-after-apply that happens in the framework and `ImportStateVerify`.